ms2query/ms2library.py

```diff
@@ -235,23 +236,27 @@ def _get_analog_search_scores(self,
             spectrum_id = query_spectrum.get(
                 self.settings["spectrum_id_column_name"])
             query_spectrum_parent_mass = query_spectrum.get("parent_mass")
+            results_table = ResultsTable(preselection_cut_off=preselection_cut_off,
+                                         spectrum_id=spectrum_id,
+                                         parent_mass=query_spectrum_parent_mass)
 
             # Select the library spectra that have the highest ms2ds with the
             # query spectrum and calculate the average and related ms2ds scores
-            selected_spectrum_ids, average_ms2ds_scores, \
-                closely_related_inchikey_scores = \
+            results_table = \
                 self._calculate_averages_and_preselection(
+                    results_table,
                     ms2ds_scores[spectrum_id],
                     preselection_cut_off)
+            results_table.data = results_table.data.set_index('spectrum_ids')
 
-            s2v_scores = self._get_s2v_scores(query_spectrum,
-                                              selected_spectrum_ids)
+            results_table.data["s2v_scores"] = self._get_s2v_scores(query_spectrum,
+                                                                    results_table.data.index.values)
+
+            parent_masses = np.array([self.parent_masses_library[x] for x in results_table.data.index])
+            results_table.add_parent_masses(parent_masses,
+                                            self.settings["base_nr_mass_similarity"])
 
-            dict_with_preselected_spectra_info[spectrum_id] = \
```

Comment:
select_and_normalize_scores_for_selected_spectra can be removed, since it is now replaced by ResultClass

Reply:
It cleans up the confusing functions and long variables really nicely!